storagenode/collector/service_test.go

```diff
@@ -61,9 +64,37 @@ func TestCollector(t *testing.T) {
 			require.NoError(t, err)
 			require.Equal(t, int64(0), used)
 
+			// ensure we haven't deleted used serials
+			err = usedSerials.IterateAll(ctx, func(_ storj.NodeID, _ storj.SerialNumber, _ time.Time) {
+				serialsPresent++
+			})
+			require.NoError(t, err)
+
 			collections++
 		}
 
 		require.NotZero(t, collections)
+		require.Equal(t, serialsPresent, 2)
+
+		serialsPresent = 0
+
+		// imagine we are 48 days in the future
+		for _, storageNode := range planet.StorageNodes {
+			usedSerials := storageNode.DB.UsedSerials()
+
+			// collect all the data
+			err = storageNode.Collector.Collect(ctx, time.Now().Add(48*24*time.Hour))
```

Comment:
I know this is tests, but should we try to use `UTC` time across the board?

Reply:
I'm honestly not sure. Initially I was thinking that using UTC in DB will avoid problems, however I think datetime without timezone created some problems, I didn't know about. So I currently just used with timezone and let DB handle it. But this UTC and no timezone vs non-UTC + timezone needs to be revisited.

Reply:
“with timezone” will store a unix time, essentially, so it might as well be UTC. The benefit is that no code on the setting or fetching sides needs to remember to do UTC conversions. See https://wiki.postgresql.org/wiki/Don't%5FDo%5FThis#Don.27t_use_timestamp_.28without_time_zone.29

Reply:
Note this is sqlite code.

Reply:
Also we have https://storjlabs.atlassian.net/browse/V3-1882

Reply:
My point was intended to apply in general, not just here.